shared/metadata.go

```diff
@@ -70,35 +89,34 @@ func parseMetadata(metadataByteMap map[string][]byte, log Logger) map[string]Met
 		var metadata Metadata
 		err := yaml.Unmarshal(data, &metadata)
 		if err != nil {
-			log.Warningf("Failed to unmarshal %s.yml.", path)
+			log.Warningf("Failed to unmarshal %s.", path)
 			continue
 		}
 		metadataMap[path] = metadata
 	}
 	return metadataMap
 }
 
-// ConstructMetadataResponse constructs the response to a WPT Metadata query.
-// When parsing 'link' nodes, assume there is no mising information nor duplicates;
-// assume each test for each browser type is only associated with one bug.
-func constructMetadataResponse(testRuns []TestRun, metadata map[string]Metadata) MetadataResults {
+// constructMetadataResponse constructs the response to a WPT Metadata query, given ProductSpecs.
+func constructMetadataResponse(productAtRevisions []ProductAtRevision, metadata map[string]Metadata) MetadataResults {
```

Comment:
Use `ProductSpec` not `ProductAtRevision` (the latter drops labels, which can be important).

Add a function for TestRun to get the `ProductSpec` (ProductSpec uses a set of labels, TestRun uses an array. Annoying, I know).